File: Flutter/xkcd_viewer/lib/models/xkcd_comic.dart
class XkcdComic {
  final String title;
  final String altText;
  final String imgUrl;
  final int comicNumber;
  final int year;
  final int month;
  final int day;

  XkcdComic({
    required this.title,
    required this.altText,
    required this.imgUrl,
    required this.comicNumber,
    required this.year,
    required this.month,
    required this.day,
  });

  factory XkcdComic.fromJson(Map<String, dynamic> json) {
    return XkcdComic(
      title: json['title'],
      altText: json['alt'],
      imgUrl: json['img'],
      comicNumber: json['num'],
      year: int.parse(json['year']),
      month: int.parse(json['month']),
      day: int.parse(json['day']),
    );
  }
}

File: Uno/XkcdViewer/XkcdViewer.Shared/Model/XkcdComic.cs
using System.Text.Json.Serialization;

namespace XkcdViewer;

public class XkcdComic
{
    [JsonPropertyName("month")]
    public string Month { get; set; }

    [JsonPropertyName("num")]
    public int Num { get; set; }

    [JsonPropertyName("link")]
    public string Link { get; set; }

    [JsonPropertyName("year")]
    public string Year { get; set; }

    [JsonPropertyName("news")]
    public string News { get; set; }

    [JsonPropertyName("safe_title")]
    public string SafeTitle { get; set; }

    [JsonPropertyName("transcript")]
    public string Transcript { get; set; }

    [JsonPropertyName("alt")]
    public string Alt { get; set; }

    [JsonPropertyName("img")]
    public string Img { get; set; }

    [JsonPropertyName("title")]
    public string Title { get; set; }

    [JsonPropertyName("day")]
    public string Day { get; set; }
    public string DateDisplay => $"{Year}-{Month}-{Day} ({Num})";
}
